src/Template/Element/top_menu.ctp

```diff
@@ -178,14 +187,8 @@ $sidenavPosition = $htmlDir == 'rtl' ? 'md-sidenav-right' : 'md-sidenav-left';
         <div id="user_menu" ng-cloak>
             <?php
             // User menu
-            if (!$session->read('Auth.User.id')) {
-                $action = $this->request->getParam('action');
-                $controller = $this->request->getParam('controller');
-                $isOnLoginPage = ($controller == 'Users' && $action == 'login');
-
-                if (!$isOnLoginPage) {
-                    echo $this->element('login');
-                }
+            if (!$session->read('Auth.User.id') && !$isOnLoginPage) {
+                echo $this->element('login');
             } else {
                 echo $this->element('space', ['htmlDir' => $htmlDir]);
             }
```

Comment:
This change is the reason for #2487.

The problem is that there are three cases and not just two:
1. User is logged in -> display the `space` element in the top menu
2. User is not logged in and is on the login page -> display neither the `space` element nor the `login` element
3. User is not logged in and on any other page -> display the `login` element

So I guess you want something like
```php
if ($session->read('Auth.User.id')) {
    echo $this->element('space', ['htmlDir' => $htmlDir]);
} elseif (!$isOnLoginPage) {
    echo $this->element('login');
}
```